### ISSUE TITLE
Ensure mapnik::value implements =,!=,<,<=,>,>= for all possible pre-mutations of value types

### ISSUE THREAD

**Issue by artemp** [mapnik/mapnik#3161]
this is fails with current master : 

```c++
 mapnik::value v1 = 1.01;
 mapnik::value v2 = true;
REQUIRE( v1 > v2 )
 ```
due to lack of specialisation! It's very easy to make a mistake so we need to either come up with a different approach or at lease ensure all required specialisations are present

**Comment by artemp** [comment]
Lets revisit mapnik::value comparison implementation 
http://www.drdobbs.com/cpp/its-hard-to-compare-floating-point-numbe/240149806

**Comment by artemp** [comment]
fixed in 2893ee4551052da3aba1a54c63e0883ad6601855

**Comment by artemp** [comment]
The current implementation doesn't comply with C++ order-relation requirements but `good-enough` for our purpose. Closing with a view to re-visit at a later stage.
